src/BinaryParsers/PEBinary/PortableExecutable/PE.cs

```diff
@@ -957,5 +965,97 @@ public bool IncrementalLinkingEnabled(Pdb pdb)
+                {
+                    // There should be exactly one symbol in the binary containing knowledge about the linker command-line.  If that one symbol
+                    // says that OPT:ICF was enablelsd then the whole PE had it enabled too.
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        public bool IsOptimizeReferencesEnabled(Pdb pdb)
+        {
+            foreach (DisposableEnumerableView<Symbol> omView in pdb.CreateObjectModuleIterator())
+            {
+                Symbol om = omView.Value;
+                if (om.GetObjectModuleDetails().OptimizeReferencesEnabled)
+                {
+                    // There should be exactly one symbol in the binary containing knowledge about the linker command-line.  If that one symbol
+                    // says that OPT:REF was enabled then the whole PE had it enabled too.
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        public bool IsLinkTimeCodeGenerationEnabled(Pdb pdb)
+        {
+            foreach (DisposableEnumerableView<Symbol> omView in pdb.CreateObjectModuleIterator())
+            {
+                ObjectModuleDetails omDetails = omView.Value.GetObjectModuleDetails();
+                if (omDetails.LinkTimeCodeGenerationEnabled)
+                {
+                    // There should be exactly one symbol in the binary containing knowledge about the linker command-line.  If that one symbol
+                    // says that LTCG was enabled then the whole PE had it enabled too.
+                    return true;
+                }
+                else if (omDetails.WholeProgramOptimization)
+                {
+                    // "If you don't explicitly specify /LTCG when you pass /GL or MSIL modules to the linker, the linker eventually detects
+                    //  this situation and restarts the link by using /LTCG."
+                    // https://docs.microsoft.com/cpp/build/reference/ltcg-link-time-code-generation?view=msvc-170#remarks
+                    //
+                    // In other words if even a single compiland has /GL enabled then LTCG was used.
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Examine the given PDB and determine whether it represents a
+        /// binary that was compiled with the Microsoft C/C++ compiler.
+        /// </summary>
+        /// <param name="pdb">The PDB.</param>
+        /// <returns>True if it looks like a MSVC binary.</returns>
+        /// <remarks>
+        /// This isn't as simple as looking for the well known compiler
+        /// values because both rust and clang binaries can link with
+        /// C runtime libraries compiled with MSVC.
+        /// </remarks>
+        public bool IsTargetCompiledWithMSVC(Pdb pdb)

```

Comment:
>IsTargetCompiledWithMSVC [](http://example.com/codeflow?start=20&length=24)

If you don't like my 'Msvc' compromise, we could use a more descriptive name here, e.g.:

IsTargetCompiledWithMicrosoftNativeCompiler
IsTargetCompiledWithMicrosoftCppCompiler

.... yes, that latter doesn't strictly account for a vanilla C compilation. :)